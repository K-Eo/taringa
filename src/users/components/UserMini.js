import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import Stat from '../../common/Stat'

export const UserMini = props => {
  const fullname = `${props.firstname} ${props.lastname}`

  return (
    <div className="col-12 col-md-6 mb-4">
      <div className="card">
        <div className="card-body p-2 d-flex">
          <img
            src={props.avatar}
            className="rounded mr-3 border"
            alt={props.username}
            style={{
              height: 68,
              maxHeight: 68,
              maxWidth: 68,
              minHeight: 68,
              minWidth: 68,
              width: 68,
            }}
          />
          <div style={{ overflow: 'hidden' }}>
            <h6
              className="m-0 text-truncate"
              title={fullname}
              style={{ fontSize: '90%' }}
            >
              {fullname}
            </h6>
            <Link
              to={`/u/${props.username}`}
              style={{ fontSize: '85%', display: 'block', lineHeight: 1.2 }}
            >
              @{props.username}
            </Link>

            <div className="mt-2">
              <Stat
                count={props.followers}
                label="seguidor"
                plural="seguidores"
              />
              <Stat
                className="ml-3"
                count={props.following}
                label="siguiendo"
                plural="siguiendo"
              />
              <Stat className="ml-3" count={props.stories} label="post" />
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state, ownProps) => {
  return state.entities.users[ownProps.id]
}

export default connect(mapStateToProps)(UserMini)
