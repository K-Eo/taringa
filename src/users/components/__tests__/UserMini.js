import React from 'react'

import { UserMini } from '../UserMini'


describe('UserMini', () => {
  it('exists', () => {
    expect(UserMini).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<UserMini />)
    expect(wrapper).toMatchSnapshot()
  })
})
