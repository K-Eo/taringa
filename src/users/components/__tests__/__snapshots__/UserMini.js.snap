// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMini renders default 1`] = `
<div
  className="col-12 col-md-6 mt-4"
>
  <div
    className="card"
  >
    <div
      className="card-body p-2 d-flex"
    >
      <div
        className="rounded mr-3 border"
        style={
          Object {
            "height": 68,
            "maxHeight": 68,
            "maxWidth": 68,
            "minHeight": 68,
            "minWidth": 68,
            "width": 68,
          }
        }
      >
        <LazyLoad
          debounce={true}
          height="auto"
          offset={100}
          once={true}
          overflow={false}
          resize={false}
          scroll={true}
          unmountIfInvisible={false}
        >
          <img
            className="img-fluid"
          />
        </LazyLoad>
      </div>
      <div
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <h6
          className="m-0 text-truncate"
          style={
            Object {
              "fontSize": "90%",
            }
          }
          title="undefined undefined"
        >
          undefined undefined
        </h6>
        <Link
          replace={false}
          style={
            Object {
              "display": "block",
              "fontSize": "85%",
              "lineHeight": 1.2,
            }
          }
          to="/u/undefined"
        >
          @
        </Link>
        <div
          className="mt-2"
        >
          <Stat
            label="seguidor"
            plural="seguidores"
          />
          <Stat
            className="ml-3"
            label="siguiendo"
            plural="siguiendo"
          />
          <Stat
            className="ml-3 d-lg-block ml-lg-0 d-xl-inline-block ml-xl-3"
            label="post"
            plural={null}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
